### PR TITLE
add client-sdk cli

### PR DESCRIPTION
**Pull request by rnpenguin**

### CODE REVIEW

File: libraries/hatch-client-sdk/src/hatch-client-sdk.ts [hunk 1]
```diff
@@ -0,0 +1,23 @@
+import {createClientSDKByDependency, createClientSDKByInputSpec} from './util';
+
+const usage = 'Usage: hatch-client-sdk [ --input input-spec | --dependency dependency-name ]';
+const argv = process.argv.slice(2);
+const typeArg = argv[0];
+if (argv.length < 2) {
+  console.error('Invalid arguments');
+  console.log(usage);
```
Comment (kiltyj) [inline libraries/hatch-client-sdk/src/hatch-client-sdk.ts:8]: Should we be throwing or exiting with an error code here and elsewhere?

File: tools/hatch-cli/src/util.ts [hunk 2]
```diff
@@ -471,6 +543,22 @@ export const createFromTemplate = async (
           fs.writeFileSync(rushConfigPath, rushConfigRawUpdated);
           await updateDockerComposition(templateName, monorepoRootDir, toShortName(name));
         }
+        if (clientSDKOptions != null) {
+          const clientSDKPackagePath = path.resolve(tempFilePath, 'package.json');
```
Comment (kiltyj) [inline tools/hatch-cli/src/util.ts:547]: might be good to refactor this out into its own method, this method is stating to get pretty long

File: tools/hatch-cli/src/templates/client-sdk/command.ts [hunk 3]
```diff
@@ -0,0 +1,16 @@
...
+
+commander
+  .name('hatch client-sdk')
+  .option('-n, --name <name>', 'The client SDK package name. Defaults to <dependency>-sdk if ' +
+    'generating from a dependency, else required', null)
+  .option('-d, --dependency <dependency>', 'The hatch microservice or webapp dependency name to ' +
+    'generate the OpenAPI spec', null)
+  .option('-v, --ver <ver>', 'The hatch microservice or webapp dependency version to generate the ' +
+    'OpenAPI spec. Defaults to \"latest\"', null)
+  .option('-i, --input <input>', 'The location of the OpenAPI spec, as URL or file', null)
```
Comment (kiltyj) [inline tools/hatch-cli/src/templates/client-sdk/command.ts:12]: per offline discussion, let's use a more descriptive name here instead of `input`

File: tools/hatch-cli/src/templates/client-sdk/template/README.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
+# HATCH_CLI_TEMPLATE_VAR_projectShortName
+This project is an autogenerated TypeScript client sdk project based on an [OpenAPI Specification](https://swagger.io/specification/).
```
Comment (kiltyj) [inline tools/hatch-cli/src/templates/client-sdk/template/README.md:2]: I'd reword to indicate that only some of the files are autogenerated. For many libraries, there might be a wrapper class that is handwritten.

File: tools/hatch-cli/src/templates/client-sdk/command.ts (hunk 3 above)
Comment (rnpenguin) [reply tools/hatch-cli/src/templates/client-sdk/command.ts]: updated

File: tools/hatch-cli/src/templates/client-sdk/template/README.md (hunk 4 above)
Comment (rnpenguin) [reply tools/hatch-cli/src/templates/client-sdk/template/README.md]: updated

File: libraries/hatch-client-sdk/src/hatch-client-sdk.ts (hunk 1 above)
Comment (rnpenguin) [reply libraries/hatch-client-sdk/src/hatch-client-sdk.ts]: updated

File: tools/hatch-cli/src/util.ts (hunk 2 above)
Comment (rnpenguin) [reply tools/hatch-cli/src/util.ts]: updated

**Review by kiltyj — APPROVED** [review]
